app/controllers/events_controller.rb

```diff
@@ -0,0 +1,55 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: [:show]
+  before_action :signed_in?, only: [:new]
+
+  # GET /events
+  # GET /events.json
+  def index
+    @events = Event.all
+  end
```

Comment:
You should list both past and upcoming events in the index action, i.e. 

`@past_events = Event.past_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
`@upcoming_events = Event.upcoming_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

You can then use these instance variables in your view.